### PR TITLE
Split DisruptionEvent into 2, start/end

### LINKED ISSUE #926: "No space left on device" should be DatabaseLogEvent Severity.CRITICAL if it happens during nodetool_enospc nemesis
We don't  need to mark as CRITICAL the error :
```
Apr 21 07:16:03 warning|  [shard 6] commitlog - Exception in segment reservation: storage_io_error (Storage I/O error: 28: filesystem error: open failed: No space left on device [/var/lib/scylla/view_hints/6/172.30.0.103/HintsLog-1-10808
6391073850791.log])
```

if it happens during nodetool_enospc nemesis. It's expected error:
```
2019-04-21 07:16:43,112 sct_events       L0336 INFO | 2019-04-21 07:16:43.101: (DatabaseLogEvent Severity.CRITICAL): type=DATABASE_ERROR node=Node longevity-2mv-backpressure-4d-VERSION-db-node-28db9f55-2 [18.212.201.126 | 172.30.0.88] (s
eed: False) line_number=174992 regex=Exception 
```

### PR DESCRIPTION
**Pull request by fruch**
also fixes #926

### CODE REVIEW

File: sdcm/nemesis.py [hunk 1]
```diff
@@ -82,9 +82,8 @@ def update_stats(self, disrupt, status=True, data={}):
         if self.tester.create_stats:
             self.tester.update({'nemesis': self.stats})
 
-    def publish_event(self, disrupt, status=True, data={}):
-        data['node'] = self.target_node
-        DisruptionEvent(name=disrupt, status=status, **data)
+    def publish_event(self, type, disrupt, status=True, data={}):
```
Comment (bentsi) [inline sdcm/nemesis.py]: why do we need `publish_event` on Nemesis class, we can just use DisruptionEvent directly
(it also annoying when we have 2 `publish_event` methods and finding method definition in PyCharm)

File: sdcm/nemesis.py [hunk 2]
```diff
@@ -883,6 +882,8 @@ def wrapper(*args, **kwargs):
         result = None
         error = None
         status = True
+        target_node = str(args[0].target_node)
+        args[0].publish_event('start', str(method.__name__), status, {'node': target_node})
```
[the remark this answers is not in the document]
Comment (fruch) [reply sdcm/nemesis.py]: doen

File: sdcm/nemesis.py (hunk 1 above)
Comment (fruch) [reply sdcm/nemesis.py]: done

File: sdcm/nemesis.py [hunk 3]
```diff
@@ -437,7 +441,9 @@ def approach_enospc():
                 self.log.error(str(details))
             return search_database_enospc(node) > orig_errors
 
-        with DbEventsFilter(type='NO_SPACE_ERROR'), DbEventsFilter(type='BACKTRACE', line='No space left on device'):
+        with DbEventsFilter(type='NO_SPACE_ERROR'), \
+                DbEventsFilter(type='BACKTRACE', line='No space left on device'), \
```
Comment (bentsi) [inline sdcm/nemesis.py]: do we really have a backtrace on No space left on device?

Comment (fruch) [reply sdcm/nemesis.py]: yes, on some error resaulting fron no disk space, there's a backtrace
here's an example from last longevity run:
```
2019-04-27 07:53:12,572 remote           L0203 DEBUG| [10.0.213.150] [stdout] Apr 27 07:53:12 warning|  [shard 10] seastar - Exceptional future ignored: storage_io_error (Storage I/O error: 28: No space left on device), backtrace:   0x7efc58
```

**Comment by fruch** [comment]
@bentsi remove the 28 from the NO_SPACE_ERROR

I think this is safe to merge

**Pull request merged**